subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/HttpClientHelper.java

```diff
@@ -183,6 +187,25 @@ public synchronized void close() throws IOException {
         }
     }
 
+    private static boolean isTimeoutException(Throwable throwable) {
+        Throwable rootCause = ExceptionUtils.getRootCause(throwable);
+        //http://hc.apache.org/httpclient-3.x/exception-handling.html
+        return rootCause instanceof ConnectTimeoutException || rootCause instanceof SocketTimeoutException;
```

Comment:
I think we should keep the blacklister implementation intact and rather make the check there.

Reply:
Ha, never mind. Looks like @bigdaz just did that: https://github.com/gradle/gradle/pull/3381